src/Avalonia.Controls/TransitioningContentControl.cs

```diff
@@ -162,10 +178,20 @@ private void HideOldPresenter()
         var oldPresenter = _isFirstFull ? _presenter2 : Presenter;
         if (oldPresenter is not null)
         {
-            oldPresenter.Content = null;
-            oldPresenter.IsVisible = false;
+          NotifyOfOldContent(oldPresenter);
+          oldPresenter.IsVisible = false;
         }
     }
+  
+    /// <summary>
+    /// Notifies that we no longer need the old content, so that any <see cref="OldContent"/>-observing ViewModel can deal with it as it needs.
+    /// </summary>
+    private void NotifyOfOldContent(ContentPresenter oldPresenter) {
+      var oldcontent = oldPresenter.Content;
+      oldPresenter.Content = null;
+      OldContent = oldcontent;
+      _OldContent = null; //We have signaled to any bound ViewModel that the old content was no longer needed when we set it in the above line, so we can now set it to null altogether - it should have already been taken care of by whatever logic the ViewModel has for observing the OldContent property.
```

Comment:
This is wrong: `OldContent` has been changed to `null` but nobody has been notified about it.

Reply:
This is done on purpose, because there is no point of notifying about the fact that the OldContent is no longer relevant. That is done when we set OldContent in the first place. I.e. When we set OldContent, observers get notified and dispose of it (or do whatever they need). At that point the OldContent is effectively gone from the TransitioningControl.

Reply:
Note: I know this has changed since your comment, but there was a point: implementing an Avalonia property implies that changes must be notified. For correctness, you can't go changing a property from A to B to C and notify only A to B and not B to C, even if you deem the C value irrelevant. For example any binding acting on it would still reference the old property value.